jpegenc/subblocks/quantizer/quantiser.py

```diff
@@ -0,0 +1,154 @@
+"""The above module is the hardware implementation
+    of quantizer top module"""
+
+from myhdl import always_seq, block, always_comb
+from myhdl import Signal, concat, modbv, intbv
+
+from .ramz import ramz
+from .quantiser_core import (quantizer, QuantOutputStream,
+                             QuantInputStream, QuantConfig)
+
+
+class QuantIDataStream(object):
+    """
+    Input datastream into the Quantizer top module
+
+    data_in : send input data into the module
+    read_addr : read the data from the input buffer
+
+    """
+    def __init__(self, width, addr):
+        self.data_in = Signal(intbv(0)[width:].signed())
+        self.read_addr = Signal(modbv(0)[addr:])
+
+
+class QuantCtrl(object):
+    """
+    Control Signals used for quantizer top module
+
+    start : signal used to start the processing of block
+    ready : asserts when block is ready to take next input
+    color_components : select Y1 or Y2 or Cb or Cr component
+
+    """
+    def __init__(self):
+        self.start = Signal(bool(0))
+        self.ready = Signal(bool(0))
+        self.color_component = Signal(intbv(0)[3:])
+
+
+class QuantODataStream(object):
+    """
+    Output data stream from quantizer module
+
+    data_out : output data sent from the output buffer
+    read_addr : read data from the output buffer
+
+    """
+    def __init__(self, width, addr):
+        self.data_out = Signal(intbv(0)[width:].signed())
+        self.read_addr = Signal(modbv(0)[addr:])
+
+
+@block
+def quantizer_top(
```

Comment:
In the `quantiser.py` python module you have `quantizer_top` myhdl.block and in `quantiser_core.py` python module you have `quantizer` myhdl.block ...